helix-vcs/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+mod git;
+use git::Git;
+
+use helix_core::Rope;
+
+#[derive(Copy, Clone, Debug)]
+pub enum LineChange {
+    Added,
+    RemovedAbove,
+    RemovedBelow,
+    Modified,
+}
+impl LineChange {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            LineChange::Added => &"+",
+            LineChange::RemovedAbove => &"-",
+            LineChange::RemovedBelow => &"_",
+            LineChange::Modified => &"~",
```

Comment:
I think we should just use the `▎` for add and modify and `▁` for blocks removed. UTF-8 characters will look better. And we can make use of colors, like green for added, yellow/orange for modified, red for removed.

Reply:
I personally prefer:

```

let g:gitgutter_sign_added = '▌'
let g:gitgutter_sign_removed = '▖'
let g:gitgutter_sign_removed_first_line = '▘'
let g:gitgutter_sign_modified = '▐'
let g:gitgutter_sign_modified_removed = '▞'
```

with
`diff.plus` #35BF86
`diff.minus` #F22C86
`diff.delta` #6F44F0

Reply:
While I like those glyphs, I'm not sure if they are very discoverable to unfamiliar users. That said, I am planning to make them them theme configurable in a separate PR.

Reply:
`▞` is confusing, I prefer `-` or line like for removed, at least it's clearer.

Reply:
You need to see how it looks visually. `▞` is specifically when an area is both deleted and then modified.

Reply:
<img width="746" alt="Screenshot 2021-07-19 at 23 23 59" src="https://user-images.githubusercontent.com/1372918/126175906-89d28ba8-a432-4b90-8280-7531783c9fdc.png">

In the first purple area, one line was deleted and replaced with a single line. In the second area, multiple lines were deleted and replaced with a single line.

Reply:
Without you explaning, I won't understand what that means.